iam/api-client/src/test/java/com/google/iam/snippets/ServiceAccountTests.java

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.iam.snippets;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.api.services.iam.v1.model.ServiceAccount;
+import com.google.api.services.iam.v1.model.ServiceAccountKey;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Random;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+
+@RunWith(JUnit4.class)
+public class ServiceAccountTests {
+
+  private ByteArrayOutputStream bout;
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+
+  private static final String NAME = "java-test-" + new Random().nextInt(1000);
+
+  private static final String EMAIL =  NAME + "@" + PROJECT_ID + ".iam.gserviceaccount.com";
+
+  private String got;
+
+  private static void requireEnvVar(String varName){
+      assertNotNull(
+        System.getenv(varName),
+        String.format("Environment variable '%s' is required to perform these tests.", varName)
+    );
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void beforeTest() {
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+    bout.reset();
+  }
+
+  @Test
+  public void testServiceAccounts() throws Exception {
```

Comment:
While these don't need to be in separate files, the tests should ideally be in separate functions. This way when a snippet fails, we know exactly which test is failing.

Reply:
I had tried to do that initially, but the tests kept failing because they need to be run sequentially. Is there a way to run them sequentially if they're in separate functions?

Reply:
Ideally you don't want to run tests in order, because you want them to be independent and hermetic. However sometimes this isn't feasible, in which case you case we've had [a few folks use this trick to order tests with JUnit](https://stackoverflow.com/a/24191972/9414803).

Reply:
Done